tests/posix/env/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(posix_common)
+
+FILE(GLOB app_sources src/*.c)
+
+target_sources(app PRIVATE ${app_sources})
+
+# For getenv_r() visibility
+target_compile_options(app PRIVATE -D_BSD_SOURCE)
+# To avoid errors due to
+# https://github.com/zephyrproject-rtos/zephyr/issues/67283
+target_compile_options(app PRIVATE -U_POSIX_C_SOURCE -D_POSIX_C_SOURCE=200112L)
```

Comment:
Given that this gives trouble until the picolibc fix is merged, and given that this is just to work around checkpatch, I would either wait for picolibc's fix, or have it defined in the source, and ask for an exception as in this case it is perfectly justified as you want to test setenv() and unsetenv()

Reply:
> Given that this gives trouble until the picolibc fix is merged

Can you clarify? There is zero trouble as far as I am aware.

> and given that this is just to work around checkpatch

Which is fine, since the checkpatch issues are false positives.

>  anyhow this PR triggers other checkpatch checks

Again, false positives (also mentioned in the summary).

> I would either wait for picolibc's fix

Why would that be necessary, specifically? It really isn't.

> have it defined in the source,

Defining it in the source is fine, but for now it creates an unnecessary amount of noise.

> ask for an exception as in this case it is perfectly justified as you want to test setenv() and unsetenv()

Ask for exceptions to support features that are already part of the release plan? That seems like a waste of many people's time.